Seek callback

danielmartinprieto opened the pull request:
Hi! Playing with a UISlider and the seek to time operation, I thought that having a callback when it has ended was a good idea, and as Apple already provides this functionality in AVPlayer, adding it to AudioPlayer has been easy.

delannoyk commented on AudioPlayer/AudioPlayer/player/extensions/AudioPlayer+Control.swift:
```diff
@@ -91,57 +91,80 @@ extension AudioPlayer {
-                toleranceBefore: toleranceBefore,
-                toleranceAfter: toleranceAfter)
-            updateNowPlayingInfoCenter()
+            if let player = self.player {
```
Could you remove the use of self in here?

delannoyk commented on AudioPlayer/AudioPlayer/player/extensions/AudioPlayer+Control.swift:
```diff
@@ -91,57 +91,80 @@ extension AudioPlayer {
+    /// - completionHandler: The optional callback that gets executed upon completion with a boolean param indicating if the operation has finished.
+    public func seekToSeekableRangeStart(padding: TimeInterval, completionHandler: ((Bool) -> Void)? = nil) {
+        guard let range = currentItemSeekableRange,
+            let player = self.player else {
```
Could you remove the use of self in here?

delannoyk commented on AudioPlayer/AudioPlayer/player/extensions/AudioPlayer+Control.swift:
```diff
@@ -91,57 +91,80 @@ extension AudioPlayer {
+    /// - completionHandler: The optional callback that gets executed upon completion with a boolean param indicating if the operation has finished.
+    public func seekToSeekableRangeEnd(padding: TimeInterval, completionHandler: ((Bool) -> Void)? = nil) {
+        guard let range = currentItemSeekableRange,
+            let player = self.player else {
```
Could you remove the use of self in here?

danielmartinprieto commented:
Hi! I've fixed those things.

Yesterday, I was testing doing crazy things and got this crash:
```
Terminating app due to uncaught exception 'NSInvalidArgumentException', reason: 'AVPlayerItem cannot service a seek request with a completion handler until its status is AVPlayerItemStatusReadyToPlay.'
```
So I've also added a check to see if the current item status is ready to play before trying to seek.

delannoyk commented:
Just so you know I didn't merge this yet because of the last change:
> So I've also added a check to see if the current item status is ready to play before trying to seek.
This is something that can cause bugs in other people apps. Event in my app, I currently let the user start seeking even though it's item didn't finish buffering and is still loading.

danielmartinprieto commented:
Mmm, I see. For the error message, I guess this could only happen if the seek operation has a callback attached, so I can change the code to do the check only when this is the case, thus working as before for you that are not passing any callback. What do you think?

delannoyk commented:
It would probably be the best option yes. Thanks!

danielmartinprieto commented:
Please, test that everything works as expected without doing any changes in your side!

delannoyk commented:
Thanks and sorry for the delay. Merging it!
